webui/react/src/types.ts

```diff
@@ -251,11 +251,16 @@ export interface TrialItem {
   url: string;
 }
 
-export interface Step {
-  endTime?: string;
-  id: number;
-  startTime: string;
+export interface Validation extends BaseStep{ // validation sub step?
+  state: RunState; // TODO CHECK
+  stepId: number;
+  metrics?: ValidationMetrics;
+}
+
+export interface Step extends BaseStep {
   state: RunState;
+  checkpoint?: Checkpoint;
+  validation?: Validation;
```

Comment:
**question:** can these properties just be part of `BaseStep`?

Reply:
as it's set now no? that would give the impression as if checkpoint or validation had sub checkpoints.